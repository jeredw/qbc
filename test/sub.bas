sub foo (bar$, baz as string, quux) static
end sub
sub bar(a, b, c%): end sub
sub baz: end sub: print "wow"
sub quux static
end sub
sub quux2: print "hi"
print "stuff": end sub
sub empty(): end sub
'fixed by ide but not matched by this grammar
'if 42 then
'  sub wow
'  end sub
'end if
'
'error:
'sub foo (a as string*40)
'end sub
