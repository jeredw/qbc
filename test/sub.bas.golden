(program 
  (sub_block sub foo 
    ( 
      (parameter_list 
        (parameter 
          (typed_id bar $)) , 
        (parameter baz as 
          (type_name_for_parameter_list string)) , 
        (parameter 
          (typed_id quux))) ) static 
    (sub_body_block \n 
      (sub_body_statement statement)) 
    (end_sub_statement end sub)) \n 
  (sub_block sub bar 
    ( 
      (parameter_list 
        (parameter 
          (typed_id a)) , 
        (parameter 
          (typed_id b)) , 
        (parameter 
          (typed_id c %))) ) 
    (sub_body_block :) 
    (end_sub_statement end sub)) \n 
  (sub_block sub baz 
    (sub_body_block :) 
    (end_sub_statement end sub : 
      (statement 
        (print_statement print 
          (print_args 
            (expr 
              (literal "wow"))))))) \n 
  (sub_block sub quux static 
    (sub_body_block \n 
      (sub_body_statement statement)) 
    (end_sub_statement end sub)) \n 
  (sub_block sub quux2 
    (sub_body_block : 
      (sub_body_statement 
        (statement 
          (print_statement print 
            (print_args 
              (expr 
                (literal "hi")))))) \n 
      (sub_body_statement 
        (statement 
          (print_statement print 
            (print_args 
              (expr 
                (literal "stuff")))))) : 
      (sub_body_statement statement)) 
    (end_sub_statement end sub)) \n 
  (sub_block sub empty 
    ( ) 
    (sub_body_block :) 
    (end_sub_statement end sub)) \n statement \n statement \n statement \n statement \n statement \n statement \n statement \n statement \n statement \n statement <EOF>)
